### ISSUE TITLE
Expose import map shim on JS API

### ISSUE THREAD

**Issue by josh-sachs** [guybedford/es-module-shims#9]
Currently, shim only supports defining import maps via dom.

`<script type="importmap-shim">`

We can `src` or `inline` the map, but there is a hard constraint on the existence of this element in the dom.

Would be nice to be able to use the shim in a web worker (which doesn't have a dom).

Basically, import the shim into a webworker and call something like `useMap(url|object)`

I understand the shim is trying to force a pattern as close to the final spec as possible... maybe the lib could detect it is in a webworker before defining/exposing the function?  Or throw an exception if used on the main thread. 

Happy to discuss further...

**Comment by guybedford** [comment]
So ideally we should position this project to roughly where things are headed.

One thing we could consider here is extending the shim to the worker constructor itself.

For example, `new WorkerShim('./path/to/module.js', { type: 'module', importMap: {...} })` or similar. I think we should make the `type: 'module'` mandatory too so that without it, a normal worker is created, that could also still access the dynamic import through `import()` being rewritten. That way `window.Worker = WorkerShim` could be a comprehensive polyfill.

This would like up with the current strawman at https://github.com/WICG/import-maps/issues/2.

Basically the `WorkerShim` constructor would just be a worker wrapper that contained eg the code:

```js
importScripts('/path/to/es-module-shims.js');
_importMap = {...};
importShim('/path/to/worker.js');
```

where the generated code above would run in a blob or data URL. (Workers work with data / blob right?).

Definitely sounds like useful directions.

**Comment by jbanety** [comment]
Hi,
Exposing utility functions could be usefull for dynamic import of different applications in a microservice environnement. 
So i can have several importmaps json files with specific dependencies for each application.

**Comment by guybedford** [comment]
We can support multiple import maps today, although how deferred import maps load is not yet specified. I've created https://github.com/guybedford/es-module-shims/issues/15 to track this.

The loading process for workers should be more like the approach above though I think.
